types/src/oauth/lib.ts

```diff
@@ -75,6 +75,11 @@ export const SnowflakeCredentialsSchema = t.type({
 export type SnowflakeCredentials = t.TypeOf<typeof SnowflakeCredentialsSchema>;
 export type ConnectionCredentials = SnowflakeCredentials;
 
+export const ApiCredentialsSchema = t.type({
```

Comment:
I'm unclear here because I seek a snowflake specific thing below? To be consistent we should use a similar constrcut no?

Reply:
Well it's a bit unclear from the code indeed. 
ConnectionCredentials are snowflake creds as per `export type ConnectionCredentials = SnowflakeCredentials;` 

But it seems like the code has been built to be more generic in the future (as we'll add more providers) so I would not get into creating one schema per provider if not necessary, right ? (we would duplicate a lot of stuff when only asking for API keys)

Reply:
`ConnectionCredentials` should become `SnowflakeCredentials | ModjoCredentials` IMO (or `ApiKeyCredentials` if we want to be generic, but @Fraggle  was not in favor of making this very generic IIRC)

Reply:
Fine for me, but then the union type should be `ConnectionCredentials` and not adhoc in the function args. Does that make sense ?

Reply:
I still think it has value. For example, let's say you have a TS type like this:
```
{
title:string,
content: string
}
```

Would you call it `TitleAndContentType` or you would use a more specific name ?

Reply:
(you can call it ApiKeyCredentials if you really want)

Reply:
Ok, I've switched a bit to have ApiKeyCredentialsSchema and 

```
export type ModjoCredentials = t.TypeOf<typeof ApiKeyCredentialsSchema>;
export type ConnectionCredentials = SnowflakeCredentials | ModjoCredentials;
```

Does that work with you @Fraggle ? 
So we can just do `FirefliesCredentials = t.TypeOf<typeof ApiKeyCredentialsSchema>;` without duplicating the schema